lib/NeuraDialect/Mapping/mapping_util.cpp

```diff
@@ -397,8 +397,36 @@ std::vector<std::pair<Operation *, int>> mlir::neura::flatten_level_buckets(
   std::vector<std::pair<Operation *, int>> result;
 
   for (int level = 0; level < static_cast<int>(level_buckets.size()); ++level) {
-    for (Operation *op : level_buckets[level]) {
-      result.emplace_back(op, level);
+    // Collect ops with their current index to ensure stable sorting.
```

Comment:
`// Collect` -> `// Collects`, and all other comments' first verb, e.g., `// Sort` -> `// Sorts`.